jerry-core/ecma/operations/ecma-comparison.c

```diff
@@ -393,11 +393,15 @@ ecma_op_abstract_relational_compare (ecma_value_t x, /**< first operand */
 
       if (ECMA_IS_VALUE_ERROR (ecma_op_to_number (px, &nx)))
       {
+        ecma_free_value (prim_second_converted_value);
+        ecma_free_value (prim_first_converted_value);
         return ECMA_VALUE_ERROR;
       }
 
       if (ECMA_IS_VALUE_ERROR (ecma_op_to_number (py, &ny)))
       {
+        ecma_free_value (prim_second_converted_value);
+        ecma_free_value (prim_first_converted_value);
```

Comment:
What about setting ret_value to ECMA_VALUE_ERROR and jump to the end.
Maybe
```
ret_value = ECMA_IS_VALUE_ERROR (ecma_op_to_number (px, &nx));
if (ECMA_IS_VALUE_ERROR (ret_value))
{
  goto error;
}
```

Reply:
In my opinion we should avoid goto whenever possible. In this case we can do easily, I can't see a reason to prefer the "evil" goto here.

Reply:
One reason would be code size.

Reply:
I checked, this change increases .text size with 16 bytes. Are you guys sure the ugly goto solution would decrease the code size significantly?

Reply:
+1 for the goto solution, also, the checks can be merged:

```c
if (ECMA_IS_VALUE_ERROR (ecma_op_to_number (px, &nx)) || ECMA_IS_VALUE_ERROR (ecma_op_to_number (py, &ny)))
{
  goto error;
}
```

Reply:
There's nothing ugly about using goto, we're using it all over the code base already, it works very well for cleaning up values after an error happens. It keeps the code tidy by not having to duplicate cleanup code after each error check, and if done correctly makes the code more readable.

I think this fear of using gotos is completely unwarranted. Of course you should not throw them around haphazardly, but when used correctly they can be very efficient.

Reply:
I can't understand your admiration for goto and I still think it is ugly and should be avoided whenever possible. But you are the reviewers not me, so I understanded I have to add gotos to the code to be able to fix this memory leak, I'll update the PR.

Just a note, with goto solution, the code size will be exactly same as the current size. The merged ifs didn't change anything, the compiler already merged them, but I agree it makes the code cleaner, so I merged them.